app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/DeleteFiles.kt

```diff
@@ -41,29 +41,21 @@ data class DeleteFiles(private val booksOnDiskListItem: List<BookOnDisk>) :
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as ZimManageActivity).cachedComponent.inject(this)
 
-    var name = "\n"
-    booksOnDiskListItem.forEach {
-      name += "\n" + it.book.title
-    }
+    val name = booksOnDiskListItem.joinToString(separator = "\n") { it.book.title }
 
     dialogShower.show(DeleteZims(name), {
-      var allDeleted = true
-
-      booksOnDiskListItem.forEach {
-        if (deleteSpecificZimFile(it)) {
-          if (it.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
+      val booksDeleted = booksOnDiskListItem.fold(true) { _, book ->
```

Comment:
you need to be using the accumulator, it is what is storing the result of each execution. Right now this just returns the result of the last delete and ignores the other operations, the [kotlin docs](https://kotlinlang.org/docs/reference/collection-aggregate.html#fold-and-reduce) might explain it better than me 

basically what I want to be written is
```
deleteSpecificZimFile(books[0]) && deleteSpecificZimFile(books[1]).... deleteSpecificZimFile(books[i])
```

but with a fold so it is cool, I would also use an `also` here for the side effect of setting the zimFile null